Nhse o34 orkv.i38 docsanddefaults

martinsumner opened the pull request:
Doc update and default changes

martinsumner committed:
Create InitialDesignDecisions.md

martinsumner committed:
Update InitialDesignDecisions.md

martinsumner committed:
Update InitialDesignDecisions.md

martinsumner committed:
Update InitialDesignDecisions.md

martinsumner committed:
Update InitialDesignDecisions.md

martinsumner committed:
Update InitialDesignDecisions.md

martinsumner committed:
Merge branch 'nhse-o34-orkv.i38-docsanddefaults' of https://github.com/OpenRiak/riak_kv into nhse-o34-orkv.i38-docsanddefaults

martinsumner committed:
Merge branch 'openriak-3.4' into nhse-o34-orkv.i38-docsanddefaults

martinsumner committed:
Non-controversial default changes

As per https://github.com/orgs/OpenRiak/discussions/26

martinsumner commented:
Other changes in https://github.com/orgs/OpenRiak/discussions/26 - are pending further discussion

martinsumner committed:
Test should show dvv_enabled=true

WarpEngineer commented:
Hey @martinsumner I'm not convinced we should make these breaking changes going from 3.2 to 3.4.  It's not a big enough version change for that.  I think we should save this for version 4.0.

martinsumner commented:
The ones I've added to this PR so far, I thought, were the relatively non-controversial changes (compared to changing the `allow_mult` or `notfound_ok` defaults).

The `handoff_deletes` is a recent addition, which was required for one Riak user, but kept as default of off when it was introduced until it has been through production proving with that user.

Also there's the question of timescales, as to when a major (e.g. 4.0) version will actually be available.

Perhaps one to discuss at the next catch-up, I'm open to be persuaded.

WarpEngineer commented:
Yeah, let's discuss again at the catch-up.

martinsumner committed:
Reverted dvv_enabled default update

Change to be made in Riak 4.0

martinsumner commented:
@WarpEngineer - I've reverted out the `dvv_enabled` change (and updated the linked issue to indicate this will be done in 4.0).  Ar you OK to go forward with just the change to handoff_deletes and the doc update?
